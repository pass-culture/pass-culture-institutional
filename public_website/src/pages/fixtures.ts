import { HomeData } from '@/pages/about.page'

export const homeDataFixtures: HomeData[] = [
  {
    attributes: {
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla euismod, nisl eget ultricies ultrices, nunc nisl aliquam nunc, vitae aliquam nisl nunc eu',
      name: 'Lorem ipsum dolor sit amet',
      createdAt: '2023-08-15T15:00:00.000Z',
      updatedAt: '2023-08-15T17:00:00.000Z',
      publishedAt: '2023-08-15T18:00:00.000Z',
    },
    id: 42,
  },
]
