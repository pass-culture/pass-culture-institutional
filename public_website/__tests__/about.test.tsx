import React from 'react'
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import About, { getStaticProps } from '../src/pages/about'
import { act, render, screen, waitFor } from '.'
import { restaurantDataFixtures } from './fixtures'

describe('About page', () => {
  it('should pass axe accessibility tests', async () => {
    const { container } = render(<About restaurants={[]} />)

    await waitFor(() => {
      expect(screen.queryByText('Chargement...')).toBeFalsy()
    })

    let a11yResult
    await act(async () => {
      a11yResult = await axe(container)
    })

    expect(a11yResult).toHaveNoViolations()
  })

  it('should render the page', async () => {
    const { container } = render(<About restaurants={[]} />)

    await waitFor(() => {
      expect(screen.queryByText('Chargement...')).toBeFalsy()
    })

    expect(container).toMatchSnapshot()
  })

  it('should render the page with the response of the server', async () => {
    process.env = { ...process.env, ID_TOKEN: 'your_dummy_token' }

    const { props } = await getStaticProps()
    render(<About {...props} />)
    expect(screen.queryByText('Tacos de Lyon')).toBeTruthy()
  })

  it('should render the page with restaurants', async () => {
    const { container } = render(<About restaurants={restaurantDataFixtures} />)

    await waitFor(() => {
      expect(screen.queryByText('Chargement...')).toBeFalsy()
    })

    expect(container).toMatchSnapshot()
  })

  it('should show 404 when restaurants are falsy', async () => {
    const { container } = render(<About restaurants={undefined} />)

    await waitFor(() => {
      expect(screen.queryByText('404')).toBeTruthy()
    })

    expect(container).toMatchSnapshot()
  })
})
