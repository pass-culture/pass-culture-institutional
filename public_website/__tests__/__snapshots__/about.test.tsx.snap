// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About page > should render the page 1`] = `
<div>
  <div
    class="sc-gtsqUy qFqbA"
  >
    <main
      class="sc-bdnyFh cOrhoO"
    >
      <h1
        class="sc-gKAaef ljqyJc"
      >
        About Page
      </h1>
      <p
        class="sc-iCoHzw jGcKYw"
      >
        Get started by editing
        <code
          class="sc-jSFipO cMRfmt"
        >
          pages/about.tsx
        </code>
      </p>
      <div
        class="sc-dlniIP RElFz"
      >
        <div
          class="sc-hKFymg jUOIRL"
        >
          <a
            class="sc-eCAqax jVKTjr"
            href="/"
          >
            ← Go Back
          </a>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`About page > should render the page with restaurants 1`] = `
<div>
  <div
    class="sc-gtsqUy qFqbA"
  >
    <main
      class="sc-bdnyFh cOrhoO"
    >
      <h1
        class="sc-gKAaef ljqyJc"
      >
        About Page
      </h1>
      <p
        class="sc-iCoHzw jGcKYw"
      >
        Get started by editing
        <code
          class="sc-jSFipO cMRfmt"
        >
          pages/about.tsx
        </code>
      </p>
      <h1
        class="sc-gKAaef ljqyJc"
      >
        Tacos de Lyon
      </h1>
      <p
        class="sc-iCoHzw jGcKYw"
      >
        Tacos de Lyon, c’est l’histoire de deux amis d’enfance, passionnés de cuisine, qui ont décidé de se lancer dans l’aventure de la restauration. Leur objectif : faire découvrir la gastronomie lyonnaise à travers un plat emblématique de la ville : le tacos. Pour cela, ils ont imaginé une carte de tacos à base de produits frais et de qualité, avec des recettes originales et savoureuses. Le tout, dans une ambiance conviviale et chaleureuse.
      </p>
      <div
        class="sc-dlniIP RElFz"
      >
        <div
          class="sc-hKFymg jUOIRL"
        >
          <a
            class="sc-eCAqax jVKTjr"
            href="/"
          >
            ← Go Back
          </a>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`About page > should show 404 when restaurants are falsy 1`] = `
<div>
  <div>
    404
  </div>
</div>
`;
